framework/src/org/checkerframework/framework/type/DefaultInferredTypesApplier.java

```diff
@@ -61,9 +61,15 @@ private void apply(
             AnnotationMirror inferred,
             TypeMirror inferredTypeMirror,
             AnnotationMirror top) {
-
         AnnotationMirror primary = type.getAnnotationInHierarchy(top);
         if (inferred == null) {
+            if (inferredTypeMirror.getKind() == TypeKind.WILDCARD) {
```

Comment:
Would it make sense to move this input validation up to the public method `applyInferredType`?
Also there are two places further down that check whether `inferredTypeMirror.getKind() != TypeKind.TYPEVAR`; could that validation also be done once in the public method? Could you use a more specific subtype of TypeMirror?
If not, it would be good to document why these properties do not need to hold in the other locations.